providers/jackson2/src/main/java/org/jboss/resteasy/plugins/providers/jackson/ResteasyJackson2Provider.java

```diff
@@ -72,16 +72,15 @@ public boolean isWriteable(Class<?> aClass, Type type, Annotation[] annotations,
 
    private static class ClassAnnotationKey
    {
-      private AnnotationArrayKey annotations;
+      private  Annotation[] annotations;
       private ClassKey classKey;
       private int hash;
 
       private ClassAnnotationKey(final Class<?> clazz, final Annotation[] annotations)
       {
-         this.annotations = new AnnotationArrayKey(annotations);
          this.classKey = new ClassKey(clazz);
-         hash = this.annotations.hashCode();
-         hash = 31 * hash + classKey.hashCode();
+         this.annotations = annotations;
```

Comment:
We should copy the array here since arrays are mutable. 
```
this.annotations = Arrays.copyOf(annotations, annotations.length);
```

There should be some handling for a `null` array too.

Reply:
The annotation arrays won't be changed after it calls into ResteasyJackson2Provider. This annotaitons arrays are actually fixed and filled with method annotaitons, so we don't need to do another array copy and null checks.